app/routines/update_user_contact_info.rb

```diff
@@ -46,11 +46,14 @@ def call
 
       user.salesforce_contact_id = sf_contact.id
 
-      SecurityLog.create!(
-        user: user,
-        event_type: :user_contact_id_updated_from_salesforce,
-        event_data: { contact_id: sf_contact.id }
-      )
+      if user.salesforce_contact_id.blank?
+        user.salesforce_contact_id = sf_contact.id
+      elsif user.salesforce_contact_id != sf_contact.id
+        user.salesforce_contact_id = sf_contact.id
+        warn("Updating a contact ID (#{sf_contact.id}) on an account (#{user.id}) which is different than what was previously linked.")
+      else
+        warn("Unable to update contact ID (#{sf_contact.id}) on an account (#{user.id}). Check user account security log for details.")
```

Comment:
I think this code runs for all users who have SF contacts, every time, even if there is nothing to update, so this warn will be printed a lot and not be useful.

Reply:
Changed this to only log when the id changes. I expect a lot of this code can be simplified once we get these accounts syncing with Salesforce.